plumbum/cli/application.py

```diff
@@ -320,8 +317,39 @@ def _parse_args(self, argv):
                 else:
                     swfuncs[swinfo.func] = SwitchParseInfo(swname, (val,), index)
 
+        # Extracting arguments from environment variables
+        envindex = 0
+        for env, swinfo in self._switches_by_envar.items():
+            envindex-=1
```

Comment:
SwitchParseInfo requires index, which later is used for sorting, I think that using same index as before will be wrong, command line arguments and environment variables arguments are not really same indexes. But since it's used later for sorting(and probably some other things that I did not notice) -  indexes are required, so negative is like "pre-command line " indexes
But maybe you have another idea how to handle this